packages/gatsby/src/query-runner/query-compiler.js

```diff
@@ -50,7 +50,8 @@ class Runner {
   }
 
   async parseEverything() {
-    let files = await globp(`${this.baseDir}/**/*.js`)
+    let files = await globp(`${this.baseDir}/**/*.+(t|j)s?(x)`)
+    files = files.filter(d => !d.match(/(\.stories\.|\.test\.|\.d\.)/))
```

Comment:
It's based on classic patterns of jest/storybook/typescript declaration files 😉 `ignorePath` can be a cool options but I don't want to provide useless option as it's commun patterns.

@KyleAMathews What do you thing on this issue?